core/src/main/java/de/danoeh/antennapod/core/service/FeedUpdateWorker.java

```diff
@@ -74,18 +68,31 @@ public Result doWork() {
                 if (!feed.getPreferences().getKeepUpdated()) {
                     itr.remove();
                 }
+                if (!feed.isLocalFeed()) {
+                    allAreLocal = false;
+                }
             }
             Collections.shuffle(toUpdate); // If the worker gets cancelled early, every feed has a chance to be updated
-            refreshFeeds(toUpdate, false);
         } else {
-            toUpdate = new ArrayList<>();
             Feed feed = DBReader.getFeed(feedId);
             if (feed == null) {
                 return Result.success();
             }
-            toUpdate.add(feed);
-            refreshFeeds(toUpdate, true);
+            if (!feed.isLocalFeed()) {
+                allAreLocal = false;
+            }
+            toUpdate = Collections.singletonList(feed);
+            force = true;
         }
+
+        if (!getInputData().getBoolean(FeedUpdateManager.EXTRA_EVEN_ON_MOBILE, false) && !allAreLocal) {
+            if (!NetworkUtils.networkAvailable() || !NetworkUtils.isFeedRefreshAllowed()) {
+                Log.d(TAG, "Blocking automatic update: no wifi available / no mobile updates allowed");
```

Comment:
Shouldn't this say 'and' (&) instead of 'or' (/)?

Reply:
There is no point in trying to update if there is no network. It should also not be tried if there is network and it is not allowed. Together, that's "no network **or** not allowed"

Reply:
>  if there is no network

Ok, I understand (I think); if there is no network connection at all (WiFi or mobile) _or_ if it's not allowed.

But while here you write 'no network' the code comment says no WiFi (which is different). Following the code comment: if there is no WiFi and mobile updates are allowed, the outcome with the or-based equation would be negative (downloads would be blocked, which is not the case).

If I'm still missing something maybe you can explain at the community meeting 😋

Reply:
I didn't change anything in that condition and it was like that since years, so I'm pretty sure it's fine ;)

Reply:
I'm not saying anything about the condition being fine or not (I also trust it is); only about the comment and if it represents the actual situation.